### ISSUE TITLE
Nullpointer in Creative messing with Turbines

### ISSUE THREAD

**Issue by metnavman** [ReikaKalseki/Reika_Mods_Issues#1347]
Trying to determine if this is a mod interaction issue, as the crash log is filled with crap I've not encountered before ( What the hell is a Class transformer null safety: found misbehaving transformers: !?).

http://pastebin.com/ecnDZ9Hz

While messing around in Creative with a reactor design, I stumbled across this repeatable bug that occurs whenever I added a new turbine to a steam line. The turbine would spool up with plenty of steam in the line (running 6 currently), but a Dynamometer hooked to it would only read about 100Mw of power, instead of the full amount. Same occurred when attaching an RF Gen to it. Full amount not being output.

Upon shutting down and reloading the game, I got the pasted crash and the whole thing shuts down (back to ATL launcher window). Upon fully reloading and entering the game, all Dynamometers now read full output, and RF Gen is putting out maximum power. This cycle occurs with every new turbine added to the line, with or without adding a Dynamometer, as well as breaking all of them prior to logging out. Whatever that MathTransformer line in the crash file is (to my laymen eyes), it appears to be some form of failure to properly update the turbine. 

I love the weird ones! 

Edit: BetterFPS perhaps...

Edit: Nope. Here's the same crash but a different end result with BetterFPS removed.

http://pastebin.com/2bgaKWwg

Now that MathTransformer thing is gone and it says "all safe". Whatever causes it is cleared out when the game reloads and it goes back to a "stable" state until adding another turbine and the game does "initial" calculations on that turbine. Definitely still happening though.

**Comment by ReikaKalseki** [comment]
Both crashes are from world construction. As for the dynamometer output, is the turbine *actually* at max power?

**Comment by metnavman** [comment]
Absolutely confirm that they are at 100% power. Like I said, after the crash and reload, they start reading properly.

http://pastebin.com/pFrYyrmF

Just got this with no turbines, so actually seems to be the Fission Reactor itself. Whatever it is, it's occurring in a test creative superflat world. I do not have this issue on my normal save where a fission reactor has been making Tritium for prolly 4+ hours now with multiple login/logouts and dimension changes.

**Comment by metnavman** [comment]
I'm not even convinced at this point that it's an issue with the mods in general. I'm going to test just loading a new creative super-flat world, closing it, and then re-opening and see if it errors out with nothing there. 

Edit: Just got done doing that. No errors. The crash begins once a reactor is turned created and turned on.

Edit: Further testing in a newly-created Survival world yields the same issue. Still no issue with my original world. The only variable being Thermal Dynamics/Expansion/Foundation having been installed. I will uninstall those and see if it persists.

**Comment by metnavman** [comment]
http://pastebin.com/rWd9R9ZK

So there's a new one messing around with this. Straight up DragonAPI. Attempting to start a new creative world to further mess with this reactor/turbine thing.

**Comment by ReikaKalseki** [comment]
This looks like #1323.

**Comment by metnavman** [comment]
That's really weird. That paste I just put in was a _brand new_ creative super-flat world loading up. First load attempt; nothing built.

Even with the NullPointer crashes, the reactor in question is only 13 cores and somewhere between 20-30 steam boilers in a 9x5 area. Nothing even remotely on the scale described in 1323. 

It's still only ever in any new worlds I try to make to fiddle with things. My first world hasn't had any of these issues (minus a steam problem which I think was more with the lines/grates but I solved anyways so non-issue). Blah, let me rip those Thermal Expansion mods out and load one. That's the ONLY variable between creating new worlds and the original save. Back in 10-15.

Edit: It's Thermal Expansion/Dynamics/Foundation. Removing those 3 lets a new creative world be made, fission reactors be placed and started, and then doesn't crash when shutting down and reloading the world. Looks like that's all she wrote! Whatever it is comes from generating new worlds when those mods are installed. My current game having been generated before I installed those mods, that problem isn't present. Guess that'll happen trying to add mods to this pack that were last updated August of last year.
